core/src/subgraph/provider_with_names.rs

```diff
@@ -9,45 +9,109 @@ pub struct SubgraphProviderWithNames<P, S> {
     logger: slog::Logger,
     provider: Arc<P>,
     store: Arc<S>,
-}
-
-impl<P, S> Clone for SubgraphProviderWithNames<P, S> {
-    fn clone(&self) -> Self {
-        Self {
-            logger: self.logger.clone(),
-            provider: self.provider.clone(),
-            store: self.store.clone(),
-        }
-    }
+    node_id: NodeId,
+    _deployment_event_stream_cancel_guard: CancelGuard, // cancels on drop
```

Comment:
Yeah good question. I think it just felt tidy...

Reply:
How'd you feel about removing it? The tidiest code is no code 😄

Reply:
Hmmm true. Thought about this some more, and I feel like we should be making components that do actually shut down properly, even if right now we run them forever, especially if it's just a few extra lines. It will be helpful if we want to write more sophisticated integration tests in the future, for example

Reply:
Ok, that's probably a good idea.